crates/control/src/motion/stand_up_back.rs

```diff
@@ -45,53 +37,41 @@ pub struct MainOutputs {
 }
 
 impl StandUpBack {
-    pub fn new(context: CreationContext) -> Result<Self> {
+    pub fn new(_context: CreationContext) -> Result<Self> {
         Ok(Self {
             interpolator: MotionFile::from_path("etc/motions/stand_up_back_dortmund_2022.json")?
                 .try_into()?,
-            filtered_gyro: LowPassFilter::with_smoothing_factor(
-                Vector2::zeros(),
-                *context.gyro_low_pass_filter_coefficient,
-            ),
         })
     }
 
-    pub fn cycle(&mut self, context: CycleContext) -> Result<MainOutputs> {
+    pub fn compute_stand_up_back(&mut self, context: CycleContext) -> Joints<f32> {
         let last_cycle_duration = context.cycle_time.last_cycle_duration;
-        let angular_velocity = context
-            .sensor_data
-            .inertial_measurement_unit
-            .angular_velocity;
+        let condition_input = context.condition_input;
 
-        self.filtered_gyro
-            .update(Vector2::new(angular_velocity.x, angular_velocity.y));
+        context.motion_finished[MotionType::StandUpBack] = false;
 
-        if context.motion_selection.current_motion == MotionType::StandUpBack {
-            self.interpolator
-                .advance_by(last_cycle_duration, context.condition_input);
-        } else {
-            self.interpolator.reset();
-        }
+        self.interpolator
+            .advance_by(last_cycle_duration, condition_input);
 
-        context.motion_safe_exits[MotionType::StandUpBack] = false;
         if self.interpolator.is_finished() {
-            match context.motion_command {
-                MotionCommand::StandUp { facing: Facing::Up } => self.interpolator.reset(),
-                _ => {
-                    if self.filtered_gyro.state().abs()
-                        < Vector2::new(
-                            *context.gyro_low_pass_filter_tolerance,
-                            *context.gyro_low_pass_filter_tolerance,
-                        )
-                    {
-                        context.motion_safe_exits[MotionType::StandUpBack] = true;
-                    }
-                }
-            };
+            dbg!("interpolator finished");
+            context.motion_finished[MotionType::StandUpBack] = true;
         }
 
+        self.interpolator.value()
+    }
+
+    pub fn cycle(&mut self, context: CycleContext) -> Result<MainOutputs> {
+        let current_position =
+            if let MotionType::StandUpBack = context.motion_selection.current_motion {
+                self.compute_stand_up_back(context)
+            } else {
+                self.interpolator.reset();
+                dbg!("initial positions");
+                self.interpolator.initial_positions()
```

Comment:
Why does the interpolator have a "initial positions" getter? Shouldn't the interpolator be generic over the type it interpolates? Having a getter with "positions" in the name means this interpolator can only interpolate over positions?

Reply:
Agree. What do you propose? Sth. like `initial_value`?

Reply:
Would be better but let's first agree on the discussion above.